packages/stores/src/account/base.ts

```diff
@@ -309,6 +331,63 @@ export class AccountStore<Injects extends Record<string, any>[] = []> {
     return Boolean(wallet);
   }
 
+  connectedWalletSupportsChain(
+    chainId: string
+  ): IPromiseBasedObservable<boolean> | undefined {
+    if (!chainId) return undefined;
+
+    /**
+     * Retrieve the Osmosis chain wallet. Other wallets might not be connected
+     * due to lack of support or pending approval. However, Osmosis is always
+     * approved upon connecting to the app.
+     */
+    const wallet = this.getWallet(this.osmosisChainId);
+
+    if (!wallet || wallet.walletStatus !== WalletStatus.Connected) {
+      return undefined;
+    }
+
+    const id = `${wallet.walletName}_${chainId}`;
+
+    let promiseObservable = this._walletToSupportChainPromise.get(id);
+
+    if (!promiseObservable) {
+      promiseObservable = fromPromise<boolean>(wallet.supportsChain(chainId));
```

Comment:
FYI from looking close at docs, you can pass a previous promise based observable as second param to help with smooth transitions between pending states

https://github.com/mobxjs/mobx-utils#frompromise

Reply:
Thank you. I'm unsure how to apply it here, especially since the promises are designed to execute just once. Perhaps you've noticed something I missed. Should we include the second parameter?